src/main/java/com/alvarium/annotators/http/ParseResult.java

```diff
@@ -0,0 +1,159 @@
+/*******************************************************************************
+ * Copyright 2022 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package com.alvarium.annotators.http;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import com.alvarium.contracts.RequestSpecialityComponent;
+
+import org.apache.http.Header;
+import org.apache.http.client.methods.HttpUriRequest;
+
+public class ParseResult {
+  private String seed;
+  private String signature;
+  private String keyid;
+  private String algorithm;
+
+  public ParseResult() {
+  }
+
+  public String getSeed() {
+    return this.seed;
+  }
+
+  public String getSignature() {
+    return this.signature;
+  }
+
+  public String getKeyid() {
+    return this.keyid;
+  }
+
+  public String getAlgorithm() {
+    return this.algorithm;
+  }
+
+  public void buildParseResult(HttpUriRequest request)
```

Comment:
Why do you need a separate `buildParseResult` function? Couldn't this functionality be contained in the constructor? The tests all look like this:

```
ParseResult parsed = new ParseResult();
parsed.buildParseResult(request);
```

Why not 
```
ParseResult parsed = new ParseResult(request);
String x = parsed.getSeed();
String y = parsed.getSignature();
...
```
???

If you're always going to call `buildParseResult()` after constructing the object and you aren't intentionally separating instantiation from population for performance reasons, then collapse the function.

Reply:
Fixed in [dcd31ee](https://github.com/project-alvarium/alvarium-sdk-java/commit/dcd31ee4a3869875e07ccf0c28a96420e26ec365)